libs/dbus_util/tests/examples/example_sml_interface.cpp

```diff
@@ -1,25 +1,61 @@
+#include <memory>
+#include <string>
+#include <string_view>
+
 #include <boost/asio.hpp>
 #include <boost/sml.hpp>
-#include <memory>
-#include <sdbusplus/asio/connection.hpp>
 #include <sdbusplus/asio/object_server.hpp>
+
+#include <tfc/ipc.hpp>
+#include <tfc/progbase.hpp>
+
 #include <tfc/dbus/sml_interface.hpp>
-struct state_machine {
+#include <tfc/dbus/string_maker.hpp>
+
+struct running {
+  static constexpr std::string_view name{ "running" };
+};
+struct not_running {
+  static constexpr std::string_view name{ "not_running" };
+};
+
+struct control_modes {
   auto operator()() {
+    using boost::sml::event;
     using boost::sml::operator""_s;
-    using boost::sml::operator""_e;
-    return boost::sml::make_transition_table(*"init"_s + "set_stopped"_e = "stopped"_s);
+
+    auto table = boost::sml::make_transition_table(*"not_running"_s + event<running> = "running"_s,
+                                                   "running"_s + event<not_running> = "not_running"_s);
+    return table;
   }
 };
-int main() {
+
+auto main(int argc, char** argv) -> int {
+  tfc::base::init(argc, argv);
   boost::asio::io_context ctx{};
-  auto bus = std::make_shared<sdbusplus::asio::connection>(ctx);
-  auto interface =
-      std::make_shared<sdbusplus::asio::dbus_interface>(bus, std::string{ tfc::dbus::sml::tags::path }, "StateMachineName");
+
+  tfc::ipc_ruler::ipc_manager_client const ipc_client{ ctx };
```

Comment:
I would prefer to use bare bone sdbusplus asio connection, this should only be minimal example

Reply:
changing to sdbusplus asio connection but a comment which specifies that `ipc_client` can also be used.